rtl/eclrtl/eclrtl.cpp

```diff
@@ -5302,6 +5302,19 @@ void rtlAddExceptionTag(StringBuffer & errorText, const char * tag, const char *
 
 //---------------------------------------------------------------------------
 
+void rtlSubstituteEmbeddedScript(size32_t &__lenResult, char * &__result, size32_t scriptChars, const char *script, size32_t outFieldsChars, const char *outFields, size32_t searchChars, const char *search)
+{
+    StringBuffer result;
+    result.append(rtlUtf8Size(scriptChars, script), script);
+    StringBuffer outFieldsX(rtlUtf8Size(outFieldsChars, outFields), outFields);
+    StringBuffer searchX(rtlUtf8Size(searchChars, search), search);
+    result.replaceString(searchX.str(), outFieldsX.str());
```

Comment:
It was coded for ease/reliability of coding... I could no-doubt optimize it but not sure it's worth it.